### ISSUE TITLE
Training with multiple directories

### ISSUE THREAD

**Issue by raphaelmourad** [calico/basenji#157]
Hi!

I am making a dataset from multiple directories generated by basenji_data.py. Basically, a directory per genome for the same ChIP-seq to predict.

Then I train the Basenji model with only one head. My question is : how is the model trained? (everything works fine). See my code below and also the model json below.

Here I wish that sequences (and targets) are taken from the two directories and used for training. Am I right? Is it multi genome training (or is it something slightly different)?

################################################
# read datasets
from basenji import dataset

import importlib
importlib.reload(dataset)

class Options:
    tfr_train_pattern = None
    tfr_eval_pattern = None
    out_dir = "results/Basenji/"

options=Options()

data_dirs = ["data/hg38/GM12878/processed_data/GM12878_hg38_l131k_128b/"]

data_dirs = ["data/hg38/GM12878/processed_data/GM12878_hg38_l131k_128b/",
"data/hg38/GM12878/processed_data/GM12878_gorGor6_l131k_128b/",
"data/hg38/GM12878/processed_data/GM12878_panPan2_l131k_128b/"]
train_data = []
eval_data = []
test_data = []
strand_pairs = []

for data_dir in data_dirs:
    # set strand pairs
    targets_df = pd.read_csv('%s/targets.txt'%data_dir, sep='\t', index_col=0)
    if 'strand_pair' in targets_df.columns:
        strand_pairs.append(np.array(targets_df.strand_pair))

    # load train data
    train_data.append(dataset.SeqDataset(data_dir,
        split_label='train',
        batch_size=params_train['batch_size'],
        shuffle_buffer=params_train.get('shuffle_buffer', 128),
        mode='train',
        tfr_pattern=options.tfr_train_pattern))

    # load eval data
    eval_data.append(dataset.SeqDataset(data_dir,
        split_label='valid',
        batch_size=params_train['batch_size'],
        mode='eval',
        tfr_pattern=options.tfr_eval_pattern))

    # load test data
    test_data.append(dataset.SeqDataset(data_dir,
        split_label='test',
        batch_size=params_train['batch_size'],
        mode='test',
        tfr_pattern=options.tfr_eval_pattern))

import importlib
importlib.reload(dataset)

seqnn_model = seqnn.SeqNN(params_model)

from basenji import trainer

# initialize trainer
seqnn_trainer = trainer.Trainer(params_train, train_data,
                                eval_data, options.out_dir)

# compile model
seqnn_trainer.compile(seqnn_model)

seqnn_trainer.fit_keras(seqnn_model)

########################################################"
{
    "train": {
        "batch_size": 4,
        "loss": "poisson",
        "optimizer": "sgd",
        "learning_rate": 0.1,
        "momentum": 0.99,
        "train_epochs": 20,
        "patience": 8,
        "clip_norm": 2
    },
    "model": {
        "seq_length": 131072,

        "augment_rc": true,
        "augment_shift": 3,

        "activation": "gelu",
        "norm_type": "batch",
        "bn_momentum": 0.9,

        "trunk": [
            {
                "name": "conv_block",
                "filters": 64,
                "kernel_size": 15,
                "pool_size": 8
            },
            {
                "name": "conv_tower",
                "filters_init": 64,
                "filters_mult": 1.125,
                "kernel_size": 5,
                "pool_size": 4,
                "repeat": 2
            },
            {
                "name": "dilated_residual",
                "filters": 32,
                "rate_mult": 2,
                "repeat": 6,
                "dropout": 0.25
            },
            {
                "name": "conv_block",
                "filters": 64,
                "dropout": 0.05
            }
        ],
        "head": {
            "name": "final",
            "units": 1,
            "activation": "softplus"
        }
    }
}

**Comment by davek44** [comment]
I think it's just using the first dataset only. If you want to train one head for data from multiple species, you'll need to create one dataset directory with one set of tfrecords with sequences/annotations that combines all of the species. I'm pretty skeptical that this would achieve good results unless you normalized very carefully. You're pushing the package to its limits here; sorry, this isn't fully developed out yet.
